databricks_cli/stack/api.py

```diff
@@ -0,0 +1,246 @@
+# Databricks CLI
+# Copyright 2018 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+from datetime import datetime
+import time
+import six
+
+import click
+
+from requests.exceptions import HTTPError
+from databricks_cli.jobs.api import JobsApi
+from databricks_cli.version import version as CLI_VERSION
+from databricks_cli.configure.config import get_profile_from_context, get_config_for_profile
+from databricks_cli.stack.exceptions import ConfigError
+
+DEBUG_MODE = False
+_home = os.path.expanduser('~')
+MS_SEC = 1000
+STACK_STATUS_INSERT = 'deployed'
+
+# Stack Deployment Status Folder- WIP
+# STACK_DIR = os.path.join(_home, 'databricks', 'stacks', 'beta')
+
+# Resource Types
+JOBS_TYPE = 'job'
+
+# Config Outer Fields
+STACK_NAME = 'name'
+STACK_RESOURCES = 'resources'
+STACK_DEPLOYED = 'deployed'
+
+# Resource Fields
+RESOURCE_ID = 'id'
+RESOURCE_TYPE = 'type'
+RESOURCE_PROPERTIES = 'properties'
+
+# Deployed Resource Fields
+RESOURCE_PHYSICAL_ID = 'physical_id'
+RESOURCE_DEPLOY_OUTPUT = 'deploy_output'
+RESOURCE_DEPLOY_TIMESTAMP = 'timestamp'
+
+
+class StackApi(object):
+    def __init__(self, api_client):
+        self.jobs_client = JobsApi(api_client)
+        self.host = "host/"
+        if click.get_current_context(silent=True):
+            profile = get_profile_from_context()
+            config = get_config_for_profile(profile)
+            self.host = config.host
+        self.deployed_resources = {}
+        self.deployed_resource_config = {}
+
+    def _parse_config_file(self, filename):
+        """Parse the json config"""
+        parsed_conf = {}
+        with open(filename, 'r') as f:
+            parsed_conf = json.load(f)
+
+        return parsed_conf
+
+    def _generate_stack_status_path(self, stack_path):
```

Comment:
In general, it would be good to have more comments on methods. Unlike other Api such as JobsApi, methods here are complex and readers can benefit from nice inline documentations.

Reply:
Okay i’ll add description comments to all the functions